sw/device/lib/dif/dif_uart.h

```diff
@@ -0,0 +1,151 @@
+ * Initialise an instance of UART
+ *
+ * Initialise UART instance using the configuration data in @p arg2.
+ * Information that must be retained, and is required to program UART must be
+ * stored in @p arg1.
+ *
+ * @param uart UART state data
+ * @param config UART configuration data
+ * @return true if the function was successful, false otherwise
+ */
+bool dif_uart_init(dif_uart_t *uart, const dif_uart_config_t *config,
+                   uintptr_t base_address);
+
+/**
+ * Configure an instance of UART
+ *
+ * Configure UART using the configuration data in @p arg2. This operation
+ * performs fundamental UART configuration.
+ *
+ * @param uart UART state data
+ * @param config UART configuration data
+ * @return true if the function was successful, false otherwise
+ */
+bool dif_uart_configure(const dif_uart_t *uart,
+                        const dif_uart_config_t *config);
+
+/**
+ * Set the upper limit of RX characters to be queued.
+ *
+ * Set the upper limit of RX characters to be queued in the RX FIFO.
+ * Characters are received by the UART HW from into the RX FIFO.
+ *
+ * @param uart UART state data
+ * @param watermark RX FIFO depth.
+ * @return true if the function was successful, false otherwise
+ */
+bool dif_uart_set_watermark_rx(const dif_uart_t *uart,
+                               dif_uart_watermark_t watermark);
+
+/**
+ * Set the upper limit of TX characters to be queued.
+ *
+ * Set the upper limit of TX characters to be queued in the TX FIFO.
+ * Characters are transmitted by the UART HW from the TX FIFO.
+ *
+ * @param uart UART state data
+ * @param watermark TX FIFO depth
+ * @return true if the function was successful, false otherwise
+ */
+bool dif_uart_set_watermark_tx(const dif_uart_t *uart,
+                               dif_uart_watermark_t watermark);
+
+/**
+ * Transmit a single UART character (polled)
+ *
+ * Transmit a single UART character @p arg2. This operation
+ * is polled, and will loop until a character can be written into TX FIFO.
+ *
+ * @param uart UART state data
+ * @param c Character to be transmitted
+ * @return true if the function was successful, false otherwise
+ */
+bool dif_uart_send_char_polled(const dif_uart_t *uart, char c);
+
+/**
+ * Receive a single UART character (polled)
+ *
+ * Receive a single UART character and store it in @p arg2. This operation
+ * is polled, and will loop until a character can be written into RX FIFO.
+ *
+ * @param uart UART state data
+ * @param c Received character
+ * @return true if the function was successful, false otherwise
+ */
+bool dif_uart_receive_char_polled(const dif_uart_t *uart, char *c);
+
+/**
+ * Enable an UART interrupt
+ *
+ * Enable an UART interrupt specified in @p arg2.
+ *
+ * @param uart UART state data
+ * @param int_type UART interrupt type
+ * @return true if the function was successful, false otherwise
+ */
+bool dif_uart_interrupt_enable(const dif_uart_t *uart,
+                               dif_uart_interrupt_t int_type);
```

Comment:
`int_type` is a confusing name. I'd go with either `interrupt_type` or `irq_type` (even if "interrupt request" isn't pedantically correct here).

Reply:
done